src/components/LayoutTree.tsx

```diff
@@ -446,6 +452,55 @@ const TreeNode = ({
     return lastPatches?.some(patch => deleteActions.includes(patch.actions[0]))
   })
 
+  const [showLineEndFauxCaret, setShowLineEndFauxCaret] = useState(false)
+  const [showLineStartFauxCaret, setShowLineStartFauxCaret] = useState(false)
+
+  // Hide the faux caret when typing occurs.
+  editingValueStore.subscribe(() => {
```

Comment:
Can we use `editingValueStore.useEffect`, or is the raw subscribe needed for some reason?

`useEffect` handles unsubscribing, which is missing here.